controller/src/command.rs

```diff
@@ -350,6 +352,16 @@ where
 	C: NodeClient + 'static,
 	K: keychain::Keychain + 'static,
 {
+	// If 'max' spend, calculate the 'amount' as the maximum spendable wallet balance, less fees.
+	if args.max && args.amount == 0 {
+		let utxos = owner_api.retrieve_outputs(None, false, false, None)?.1;
+		let spendable = utxos.iter().filter_map(|c| match c.output.status {
```

Comment:
here's a way to find outputs eligible to spend https://github.com/mimblewimble/grin-wallet/blob/master/libwallet/src/internal/selection.rs#L563-L570
Note that it also takes into account the 'account'. This is something I haven't thought about before. I'm not sure what should happen if you set `max` and also a specific wallet account. I'm guessing the user would expect to use max from this account? on the other hand, if the account wasn't specified, it defaults to the main account which might make it unclear that it would only use the max from the main account.

Reply:
Perfect! I wondered if a method already existed for this.Thanks!

> I'm not sure what should happen if you set max and also a specific wallet account.

I think all spend commands are account specific, so IMO this one should be too. If the user specifies an account, this _should_ (it doesn't atm) spend max from that account. If the user doesn't specify an account, it should spend max from the default account. I will use the improved method you suggest, and be sure to use the account specified by the user (or the default if unspecified).

I realize I also need to add documentation for this & update the help dialog for the send command. I'll be sure to be clear about this there.

Thanks for the quick feedback!